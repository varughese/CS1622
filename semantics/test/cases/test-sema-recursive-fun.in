int foo(int x, int y)
{
    return foo(x+1, y-1);
}

void main(void)
{
    main();
}
