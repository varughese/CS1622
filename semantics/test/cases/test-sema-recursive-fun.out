[program
  [fun-declaration
    [int]
    [foo]
    [params 
      [param [int] [x]] 
      [param [int] [y]]]
    [compound-stmt 
      [return-stmt
        [call
          [foo]
          [args 
            [+ [var [x]] [1]] 
            [- [var [y]] [1]]]
        ]]
    ]
  ]
  [fun-declaration
    [void]
    [main]
    [params]
    [compound-stmt 
      [call
        [main]
        [args]
      ]
    ]
  ]
]
